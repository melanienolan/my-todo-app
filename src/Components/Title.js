import React, { Component } from 'react'; // eslint-disable-line no-unused-vars
import PropTypes from 'prop-types';

class Title extends Component {
  render() {
    let completed;
    let total;
    if (this.props.todos) {
      total = this.props.todos.length;
      completed = this.props.todos.filter(todo => todo.completed).length;
    }
    return (
      <div className="Title">
        <h1>My Todos</h1>
        <p>{completed} / {total}</p>
      </div>
    );
  }
}

Title.propTypes = {
  todos: PropTypes.array
};

export default Title;
