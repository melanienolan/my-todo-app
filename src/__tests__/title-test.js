import React from 'react';
import Title from '../Components/Title';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

it('renders without crashing', () => {
  shallow(<Title />);
});

it('renders title component', () => {
  const wrapper = shallow(<Title />);
  expect(toJson(wrapper)).toMatchSnapshot();
});

it('renders title header message', () => {
  const wrapper = shallow(<Title />);
  const welcome = <h1>My Todos</h1>;
  expect(wrapper.contains(welcome)).toEqual(true);
});
