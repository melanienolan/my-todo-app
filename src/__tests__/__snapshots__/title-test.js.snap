exports[`test renders title component 1`] = `
<header>
  <h1>
    My Todos
  </h1>
  <p>
     / 
  </p>
</header>
`;
